### DISCUSSION TITLE
Where can I download FCOS version 38.20231027.3 for OKD 4.14?

### DISCUSSION THREAD

**Discussion by ron981** [okd-project/okd#2075] [General]
Hello,

I'm trying to set up OKD 4.14.0-0.okd-2023-11-12-042703 in an air-gapped environment. According to the release notes, this version requires Fedora CoreOS (FCOS) version 38.20231027.3.

I've tried looking for this specific FCOS version through:
- Fedora mirror manager
- Official Fedora CoreOS download page
- Various Fedora mirrors

However, I cannot locate this specific version. Could you please:
1. Provide a direct link to download FCOS 38.20231027.3?
2. Or alternatively, confirm if a newer version of FCOS 38 would be compatible with this OKD release?

Thank you for your help.

**Comment by JaimeMagiera** [comment]
Hi. This isn't an issue/bug per se. The OKD Working Group only provides resources for N and N-1 releases at this time. As of last week, that means 4.17 and 4.16 (which are SCOS-based releases). The FCOS Working Group, who was creating and storing the FCOS artifacts, has likely pruned artifacts from 2023. As to whether a newer version of FCOS would work with a 4.14 installer, we have no resources to say one way or the other.

**Comment by fortinj66** [comment]
**./openshift-install coreos print-stream-json** will tell you the initial image that the installer is using.  In the case of 4.14.0-0.okd-2023-11-12-042703 it uses FCOS 38.20230609.3.0.  It also gives the URL to download from.  
 
```              "vmware": {
                    "release": "38.20230609.3.0",
                    "formats": {
                        "ova": {
                            "disk": {
                                "location": "https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/38.20230609.3.0/x86_64/fedora-coreos-38.20230609.3.0-vmware.x86_64.ova",
                                "signature": "https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/38.20230609.3.0/x86_64/fedora-coreos-38.20230609.3.0-vmware.x86_64.ova.sig",
                                "sha256": "70c8ca28cfa95cc72a7e91d681397f0b4b0cdf890b859c7cee7fc700a8ceac04"
                            }
                        }
                    }
                },
```
 
 In this case 
 
 https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/38.20230609.3.0/x86_64/fedora-coreos-38.20230609.3.0-vmware.x86_64.ova
 
 for vmware.
 
 The install will upgrade the OS to the appropriate level during the install process

**Comment by JaimeMagiera** [comment]
@fortinj66 The FCOS Working Group asked us if it was okay to prune older artifacts, which we agreed with. While it may be downloadable, we can't guarantee it will be as the pruning process proceeds. So, we'll want to avoid just telling folks without verifying.

**Comment by fortinj66** [comment]
Jaime, I validated the image existed prior to posting.   The process is the same regardless of release. 

Also, by that logic we should prune the releases on github too.
